sixtyfps_compiler/passes.rs

```diff
@@ -96,7 +98,9 @@ pub async fn run_passes(
         .chain(std::iter::once(root_component))
     {
         focus_item::resolve_element_reference_in_set_focus_calls(component, diag);
-        focus_item::determine_initial_focus_item(component, diag);
+        if Rc::ptr_eq(component, root_component) {
+            focus_item::determine_initial_focus_item(component, diag);
+        }
```

Comment:
This is done because the determine_initial_focus_item sets the setup_code, and that was before only generated into code for the root component  but now it is done for each sub component.
We could make it such that we still only process the setup_code for the root_component,  but i think it makes more sense to generate it for each component.
Now the question is if the setup_code is a good idea to contain the code that focus the first item, especially since it won't work if there are repeater involved.  But we can think about focus handling later.

Reply:
I agree, this patch makes sense.

Somehow we need to convey to the run-time library which item should receive the focus when the `Window` receives the focus the very very first time. Instead of "pushing" that, we could also make an accessor in the component vtable, but I think that's more overhead for something that's needed only "once".